test/src/folding/parsers/highlight_parser_test.dart

```diff
@@ -177,14 +180,57 @@ import java.lang.Exception;''',
           ],
           expectedInvalid: const [],
         ),
+        _Example(
+          'Java. Comment after close brace',
+          code: '''
+class MyClass {
+  void method() {
+  }// comment
+  // comment
+}''',
+          mode: java,
+          expected: const [
+            _FB(startLine: 0, endLine: 4, type: _T.braces),
+            _FB(startLine: 1, endLine: 2, type: _T.braces),
+          ],
+          expectedInvalid: const [],
+          namedSectionParser: const BracketsStartEndNamedSectionParser(),
+        ),
+        _Example(
+          'Java. Named comments',
+          code: '''
+class MyClass {
+  // [START section1]
+  // [END section2]
+}''',
+          mode: java,
+          expected: const [
+            _FB(startLine: 0, endLine: 3, type: _T.braces),
+          ],
+          expectedInvalid: const [],
+          namedSectionParser: const BracketsStartEndNamedSectionParser(),
+        ),
       ];
 
       for (final example in examples) {
         highlight.registerLanguage('language', example.mode);
         final highlighted = highlight.parse(example.code, language: 'language');
         final parser = HighlightFoldableBlockParser();
 
-        parser.parse(highlighted);
+        final sequences = SingleLineComments.byMode[example.mode] ?? [];
+
+        final commentParser = SingleLineCommentParser.parseHighlighted(
+          text: example.code,
+          highlighted: highlighted,
+          singleLineCommentSequences: sequences,
+        );
+
+        final serviceComments = ServiceCommentFilter.filter(
+          commentParser.comments,
+          namedSectionParser: example.namedSectionParser,
```

Comment:
To close match real usage, I suggest we use a named section parser on all examples. If doing so, they either should be in each example, or `BracketStartEndNamedSectionParser` can be hardcoded here since it is the only one for now.